.github/workflows/branch-status-checks.yml

```diff
@@ -0,0 +1,56 @@
+name: Branch Version Status Checks
+
+on:
+  push:
+    branches: [ "main", "7.*", "8.*" ]
+  pull_request:
+    branches: [ "*" ]
+  workflow_call:
+    # Map the workflow outputs to job outputs
+    outputs:
+      branches:
+        description: "List of target branches"
+        value: ${{ jobs.list-target-branches.outputs.matrix }}
+
+jobs:
+  list-target-branches:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.get-branch-list.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v2
+      - id: get-branch-list
+        run: echo "::set-output name=matrix::$(yq '.target_branch_list' ./detection_rules/etc/target-branches.yml)"
```

Comment:
This is a pain because it parses from a new file called `target-branches.yml`. Parsing from `stack-schema-map.yaml` has similar content but really over complicates parsing the branches. I tried different techniques across various previous commits, but the commands kept growing (and still didn't catch all edge cases).

Reply:
I'll update the documentation. Not a problem at all.